funnel/views/mixins.py

```diff
@@ -130,3 +133,75 @@ def loader(self, profile, project, section):
             ).first_or_404()
         g.profile = section.project.profile
         return section
+
+
+class DraftModelViewMixin(object):
+    def get_draft(self, current_obj=None):
+        """
+        Returns the draft object for `current_obj`. Defaults to `self.obj`.
+        `current_obj` is needed in case of multi-model views.
+        """
+        obj = current_obj or self.obj
+        return Draft.query.get((self.model.__tablename__, obj.uuid))
+
+    def delete_draft(self, current_obj=None):
+        """
+        Deletes draft for `current_obj`, or `self.obj` if `current_obj` is `None`.
+        """
+        draft = self.get_draft(current_obj)
+        if draft is not None:
+            db.session.delete(draft)
+            db.session.commit()
+
+    def autosave_get(self, current_obj=None):
+        """
+        Returns a tuple of the current draft revision and the formdata needed to initialize forms
+        """
+        draft = self.get_draft(current_obj)
+        initial_formdata = MultiDict(draft.body['form']) if draft is not None else None
+        return draft.revision, initial_formdata
+
+    def autosave_post(self, current_obj=None):
+        """
+        Handles autosave POST requests
+        """
+        if 'form.revision' not in request.form:
+            # as form.autosave is true, the form should have `form.revision` field even if it's empty
+            return {'error': _("Form must contain a revision ID.")}, 400
+        if forms.Form().validate_on_submit():
+            incoming_data = MultiDict(request.form.items(multi=True))
+            client_revision = incoming_data.pop('form.revision')
+            incoming_data.pop('csrf_token')
```

Comment:
`incoming_data.pop('csrf_token', None)`. There won't be a token if the form has CSRF disabled. This will be the case in unit tests.